hyperspy/_signals/eels.py

```diff
@@ -1433,6 +1487,211 @@ def create_model(self, ll=None, auto_background=True, auto_add_edges=True,
                           dictionary=dictionary)
         return model
 
+    def plot(self, plot_edges=False, only_edges=('Major', 'Minor'),
+             **kwargs):
+        """Plot the EELS spectrum. Markers indicating the position of the
+        EELS edges can be added.
+
+        Parameters
+        ----------
+        plot_edges : {False, True, list of string or string}
+            If True, draws on s.metadata.Sample.elements for edges.
+            Alternatively, provide a string of a single edge, or an iterable
+            containing a list of valid elements, EELS families or edges. For
+            example, an element should be 'Zr', an element edge family should
+            be 'Zr_L' or an EELS edge 'Zr_L3'.
+        only_edges : tuple of string
+            Either 'Major' or 'Minor'. Defaults to both.
+        kwargs
+            The extra keyword arguments for plot()
+        """
+
+        super().plot(**kwargs)
+
+        if plot_edges is not False:
+            edges = self._get_edges_to_plot(plot_edges, only_edges)
+            self.plot_edges_label(edges)
+
+    def plot_edges_label(self, edges, vertical_line_marker=None,
+                            text_marker=None):
+        """Put the EELS edge label (vertical line segment and text box) on
+        the signal
+
+        Parameters
+        ----------
+        edges : dictionary
+            A dictionary with the labels as keys and their energies as values.
+            For example, {'Fe_L2': 721.0, 'O_K': 532.0}
+        vertical_line_marker :  list
+            A list contains HyperSpy's vertical line segment marker, if None,
+            determine from the given edges
+        text_marker :  list
+            A list contains HyperSpy's text box marker, if None,
+            determine from the given edges
+
+        Raises
+        ------
+        ValueError
+            If the size of edges, vertical_line_marker and text_marker do not
+            match.
+        """
+
+        if vertical_line_marker is None or text_marker is None:
+            # get position of markers for edges if no marker is provided
+            slp = SpectrumLabelPosition(self)
+            vertical_line_marker, text_marker = slp.get_markers(edges)
+        if len(vertical_line_marker) != len(text_marker) or \
+            len(edges) != len(vertical_line_marker):
+            raise ValueError('The size of edges, vertical_line_marker and '
+                             'text_marker needs to be the same.')
+
+        # add the markers to the signal and store them
+        self.add_marker(vertical_line_marker + text_marker, render_figure=False)
+        added = dict(zip(edges, map(list, zip(vertical_line_marker, text_marker))))
+        self._edge_markers.update(added)
```

Comment:
When changing indices, the labels doesn't update, while they do with the interactive tool. I guess the function drawning/updating the label should be connected to the `axes_manager` here.

An example to reproduce:
```python
s = hs.datasets.artificial_data.get_core_loss_eels_line_scan_signal(True)
s.add_elements(['Cr', 'Fe'])
s.plot(plot_edges=True)
```

Reply:
Thanks for checking this! completely overlooked... now the labels will update both in static and interactive mode.